File: ios/Classes/message/OtherMessageNode.swift
import ImSDK

//
//  TextMessageNode.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/3/13.
//  其它消息节点
public class OtherMessageNode : AbstractMessageNode{

    override func getNote(elem: TIMElem) -> String {
        return "[其它消息]";
    }
    
    override func analysis(elem: TIMElem) -> AbstractMessageEntity {
        let entity = OtherMessageEntity();
        entity.params = "{\"note\":\"IOS暂不支持该节点!\"}";
        entity.type = "\(type(of:elem))";
        return entity;
    }
}

File: android/src/main/java/top/huic/tencent_im_plugin/message/OtherMessageNode.java
package top.huic.tencent_im_plugin.message;

import com.tencent.imsdk.TIMElem;
import com.tencent.imsdk.TIMMessage;

import top.huic.tencent_im_plugin.message.entity.OtherMessageEntity;
import top.huic.tencent_im_plugin.util.JsonUtil;

/**
 * 其它消息节点
 */
public class OtherMessageNode extends AbstractMessageNode<TIMElem, OtherMessageEntity> {
    @Override
    protected TIMMessage getSendMessage(OtherMessageEntity entity) {
        throw new RuntimeException("This node does not support sending");
    }

    @Override
    public String getNote(TIMElem elem) {
        return "[其它消息]";
    }

    @Override
    public OtherMessageEntity analysis(TIMElem elem) {
        OtherMessageEntity entity = new OtherMessageEntity();
        entity.setType(elem.getType().toString());
        entity.setParams(JsonUtil.toJSONString(elem));
        return entity;
    }

    @Override
    public Class<OtherMessageEntity> getEntityClass() {
        return OtherMessageEntity.class;
    }
}